Contrib/ChiralPairs/ChiralDescriptors.py

```diff
@@ -0,0 +1,391 @@
+    if weighdownShared:
+        return sum(1.0 / sharedNeighbors[a] for a in sub)
+    else:
+        return len(sub)
+    
+def getBondsSubstituent(mol, atoms):
+    bonds=[]
+    for b in mol.GetBonds():
+        a1 = b.GetBeginAtomIdx()
+        a2 = b.GetEndAtomIdx()
+        if a1 in atoms and a2 in atoms:
+            bonds.append(b.GetIdx())
+    return bonds
+
+def getNumAromaticBondsSubstituent(mol, subAtoms):    
+    return sum(1 for b in getBondsSubstituent(mol, subAtoms) 
+           if mol.GetBondWithIdx(b).GetIsAromatic())
+
+def getNumRotatableBondsSubstituent(mol, subAtoms):
+    rotatableBond = Chem.MolFromSmarts('[!$(*#*)&!D1]-&!@[!$(*#*)&!D1]')
+    matches = mol.GetSubstructMatches(rotatableBond)
+    numRotBonds=0
+    for a1,a2 in matches:
+        if a1 in subAtoms and a2 in subAtoms:
+            numRotBonds+=1
+    return numRotBonds
+
+
+substituentDescriptor = namedtuple('substituentDescriptor',
+                                   ['size','relSize','numNO','relNumNO','relNumNO_2','pathLength','relPathLength','relPathLength_2',
+                                    'sharedNeighbors', 'numRotBonds', 'numAroBonds'])    
+
+def calcSizeSubstituents(mol, subs, sharedNeighbors, maxShell):  
+    sizeDict=defaultdict()
+    numAtoms = mol.GetNumAtoms()
+    for sidx, sub in sorted(subs.items(), key= lambda x: len(x[1])):
+        size = _getSizeOfSubstituents(sub, sharedNeighbors)
+        numNOs=0
+        numShared=0
+        # determine the number of oxygen and nitrogen atoms
+        for i in sub:
+            if mol.GetAtomWithIdx(i).GetAtomicNum() in [7,8]:
+                numNOs+=1.0/sharedNeighbors[i]
+            if sharedNeighbors[i] > 1:
+                numShared+=1
+        numRotBs = getNumRotatableBondsSubstituent(mol, set(sub))
+        aroBonds = getNumAromaticBondsSubstituent(mol, set(sub))
+        # fill the substituentDescriptor tuple
+        sizeDict[sidx]=substituentDescriptor(size=size,relSize=size/numAtoms,numNO=numNOs,relNumNO=numNOs/numAtoms,
+                                             relNumNO_2=numNOs/size,pathLength=maxShell[sidx],
+                                             relPathLength=maxShell[sidx]/numAtoms,relPathLength_2=maxShell[sidx]/size,
+                                                 sharedNeighbors=numShared, numRotBonds=numRotBs, numAroBonds=aroBonds)
+    # if we have less then 4 substituents the missing ones need to be an hydrogen atoms
+    if len(sizeDict) < 4:
+        for i in range(4-len(sizeDict)):
+            sizeDict['H'+str(i)]=substituentDescriptor(size=0,relSize=0,numNO=0,relNumNO=0,relNumNO_2=0,
+                                                       pathLength=0,relPathLength=0,relPathLength_2=0,sharedNeighbors=0, numRotBonds=0,
+                                                       numAroBonds=0)
+    return sizeDict
+
+
+# Visualization of the substituents
+def visualizeSubstituentsGrid(mol, aIdx, molSize=(300,150), kekulize=True,):
```

Comment:
I don't think it's critical for merging this PR (since it's for Contrib), but I am curious why you are rolling your own SVG grid construction code instead of using the `DrawMolecules()` method of the `MolDraw2D` object. Was that missing some functionality you needed?

Reply:
I guess some functionalty was missing or I would like to have something special. Since it has been a while since I wrote this code I need to think about it :). We can discuss adding the missing pieces (if they are not too special) to the general DrawMolecules() function.

Reply:
sounds good